app/src/util/literalai_util.py

```diff
@@ -0,0 +1,105 @@
+import argparse
+import functools
+import json
+import logging
+import os
+import pickle
+import sys
+from datetime import datetime
+from typing import Any
+
+from literalai import LiteralClient, Thread
+from literalai.filter import Filter, OrderBy
+
+from src.app_config import app_config
+
+logger = logging.getLogger(__name__)
+
+
+@functools.cache
+def client() -> LiteralClient:  # pragma: no cover
+    if app_config.literal_api_key_for_api:
+        return LiteralClient(api_key=app_config.literal_api_key_for_api)
+    return LiteralClient()
+
+
+def get_project_id() -> str:
+    lai_client = client()
+    return lai_client.api.get_my_project_id()
+
+
+def get_threads(filters: list[Filter]) -> list[Thread]:
+    logger.info("Query filter: %r", filters)
+    order_by: OrderBy = OrderBy(column="createdAt", direction="ASC")
+
+    lai_client = client()
+    threads = []
+    after = None
+    while True:
+        response = lai_client.api.get_threads(filters=filters, order_by=order_by, after=after)
+        after = response.pageInfo.endCursor
+        threads += response.data
+        logger.info("Got %r of %r total threads", len(threads), response.totalCount)
+        if not response.pageInfo.hasNextPage:
+            assert (
+                len(threads) == response.totalCount
+            ), f"Expected {response.totalCount} threads, but got only {len(threads)}"
+            return threads
+
+
+def query_threads_between(start_date: datetime, end_date: datetime) -> list[Thread]:
+    filters: list[Filter] = [
+        Filter(field="createdAt", operator="gte", value=start_date.isoformat()),
+        Filter(field="createdAt", operator="lt", value=end_date.isoformat()),
+    ]
+    return get_threads(filters)
+
+
+def save_threads(threads: list[Thread], basefilename: str) -> None:  # pragma: no cover
```

Comment:
Didn't bother testing save and load functions since they are trivial (and they'd write to disk)